### PR TITLE
[PIR] fix translation for inference model: cascade_rcnn_r50_fpn_1x_coco

### PR DESCRIPTION
**Pull request by kangguangli**
### PR types

Others
### PR changes

Others
### Description

Unresolved case:
```
{ // block 8
    var _generated_var_22 : LOD_TENSOR.shape(1,).dtype(int32).stop_gradient(False)
    var _generated_var_23 : LOD_TENSOR.shape(1, 6).dtype(float32).stop_gradient(False)
    var _generated_var_24 : LOD_TENSOR.shape().dtype(int64).stop_gradient(False)
    var _generated_var_25 : STEP_SCOPES)
    var _generated_var_26 : LOD_TENSOR.shape(-1,).dtype(int32).stop_gradient(False)
    var _generated_var_27 : LOD_TENSOR.shape(-1, 6).dtype(float32).stop_gradient(False)
    var _generated_var_28 : LOD_TENSOR.shape(-1,).dtype(int64).stop_gradient(False)
    var _generated_var_29 : STEP_SCOPES)
    var _generated_var_30 : LOD_TENSOR.shape(-1,).dtype(int32).stop_gradient(False)
    var _generated_var_31 : LOD_TENSOR.shape(-1, 6).dtype(float32).stop_gradient(False)
    var _generated_var_32 : LOD_TENSOR.shape().dtype(int64).stop_gradient(False)
    var array_length_6.tmp_0 : LOD_TENSOR.shape(1,).dtype(int64).stop_gradient(False)
    var array_length_7.tmp_0 : LOD_TENSOR.shape(1,).dtype(int64).stop_gradient(False)
    var cast_16.tmp_0 : LOD_TENSOR.shape().dtype(int32).stop_gradient(False)
    var logical_not_3.tmp_0 : LOD_TENSOR.shape().dtype(bool).stop_gradient(False)
    var multiclass_nms3_0.tmp_2_slice_0 : LOD_TENSOR.shape().dtype(int32).stop_gradient(False)
    var range_11.tmp_0_slice_0 : LOD_TENSOR.shape().dtype(int64).stop_gradient(False)
    var tmp_164 : LOD_TENSOR.shape().dtype(int64).stop_gradient(False)
    var tmp_167 : LOD_TENSOR.shape().dtype(int64).stop_gradient(False)
    var tmp_168 : LOD_TENSOR.shape().dtype(int32).stop_gradient(False)
    var tmp_169 : LOD_TENSOR.shape().dtype(bool).stop_gradient(False)
    var tmp_179 : LOD_TENSOR.shape().dtype(int64).stop_gradient(False)
    var tmp_180 : LOD_TENSOR.shape().dtype(int64).stop_gradient(False)
    var tmp_181 : LOD_TENSOR.shape().dtype(bool).stop_gradient(False)

    {Out=['tmp_164']} = scale(inputs={ScaleTensor=[], X=['fill_constant_59.tmp_0']}, bias = 1.0, bias_after_scale = True, op_device = , op_namescope = /, op_role = 0, op_role_var = [], scale = 1.0)
    {Out=['range_11.tmp_0_slice_0']} = slice(inputs={EndsTensor=[], EndsTensorList=['tmp_164'], Input=['range_11.tmp_0'], StartsTensor=[], StartsTensorList=['fill_constant_59.tmp_0']}, axes = [0], decrease_axis = [0], ends = [-1], infer_flags = [-1], op_device = , op_namescope = /, op_role = 0, op_role_var = [], starts = [-1])
    {Out=['tmp_167']} = scale(inputs={ScaleTensor=[], X=['range_11.tmp_0_slice_0']}, bias = 1.0, bias_after_scale = True, op_device = , op_namescope = /, op_role = 0, op_role_var = [], scale = 1.0)
    {Out=['multiclass_nms3_0.tmp_2_slice_0']} = slice(inputs={EndsTensor=[], EndsTensorList=['tmp_167'], Input=['multiclass_nms3_0.tmp_2'], StartsTensor=[], StartsTensorList=['range_11.tmp_0_slice_0']}, axes = [0], decrease_axis = [0], ends = [-1], infer_flags = [-1], op_device = , op_namescope = /, op_role = 0, op_role_var = [], starts = [-1])
    {Out=['tmp_168']} = fill_constant(inputs={}, dtype = 2, force_cpu = False, op_device = , op_namescope = /, op_role = 0, op_role_var = [], place_type = -1, shape = [], str_value = , value = 0.0)
    {Out=['tmp_169']} = equal(inputs={X=['multiclass_nms3_0.tmp_2_slice_0'], Y=['tmp_168']}, axis = -1, force_cpu = False, op_device = , op_namescope = /, op_role = 0, op_role_var = [], with_quant_attr = False)
    {Out=['_generated_var_22', '_generated_var_24', '_generated_var_23'], Scope=['_generated_var_25']} = conditional_block(inputs={Cond=['tmp_169'], Input=['assign_73.tmp_0', 'assign_72.tmp_0', 'fill_constant_61.tmp_0']}, is_scalar_condition = True, op_device = , op_namescope = /, op_role = 0, op_role_var = [], sub_block = block[9])
    {Out=['logical_not_3.tmp_0']} = logical_not(inputs={X=['tmp_169']}, op_device = , op_namescope = /, op_role = 0, op_role_var = [], with_quant_attr = False)
    {Out=['_generated_var_27', '_generated_var_28', '_generated_var_26'], Scope=['_generated_var_29']} = conditional_block(inputs={Cond=['logical_not_3.tmp_0'], Input=['range_11.tmp_0_slice_0', 'fill_constant_61.tmp_0', 'multiclass_nms3_0.tmp_2', 'multiclass_nms3_0.tmp_0']}, is_scalar_condition = True, op_device = , op_namescope = /, op_role = 0, op_role_var = [], sub_block = block[10])
    {Out=['cast_16.tmp_0']} = cast(inputs={X=['tmp_169']}, in_dtype = 0, op_device = , op_namescope = /, op_role = 0, op_role_var = [], out_dtype = 2, use_mkldnn = False, with_quant_attr = False)
    {Out=['_generated_var_30']} = select_input(inputs={Mask=['cast_16.tmp_0'], X=['_generated_var_26', '_generated_var_22']}, op_device = , op_namescope = /, op_role = 0, op_role_var = [], with_quant_attr = False)
    {Out=['_generated_var_31']} = select_input(inputs={Mask=['cast_16.tmp_0'], X=['_generated_var_27', '_generated_var_23']}, op_device = , op_namescope = /, op_role = 0, op_role_var = [], with_quant_attr = False)
    {Out=['_generated_var_32']} = select_input(inputs={Mask=['cast_16.tmp_0'], X=['_generated_var_28', '_generated_var_24']}, op_device = , op_namescope = /, op_role = 0, op_role_var = [], with_quant_attr = False)
    {Out=['array_length_6.tmp_0']} = lod_array_length(inputs={X=['array_3.out']}, op_device = , op_namescope = /, op_role = 0, op_role_var = [], with_quant_attr = False)
    {Out=['array_3.out']} = write_to_array(inputs={I=['array_length_6.tmp_0'], X=['_generated_var_31']}, op_device = , op_namescope = /, op_role = 0, op_role_var = [], with_quant_attr = False)
    {Out=['array_length_7.tmp_0']} = lod_array_length(inputs={X=['array_4.out']}, op_device = , op_namescope = /, op_role = 0, op_role_var = [], with_quant_attr = False)
    {Out=['array_4.out']} = write_to_array(inputs={I=['array_length_7.tmp_0'], X=['_generated_var_30']}, op_device = , op_namescope = /, op_role = 0, op_role_var = [], with_quant_attr = False)
    {Out=['tmp_179']} = scale(inputs={ScaleTensor=[], X=['fill_constant_59.tmp_0']}, bias = 1.0, bias_after_scale = True, op_device = , op_namescope = /, op_role = 0, op_role_var = [], scale = 1.0)
    {Out=['tmp_180']} = cast(inputs={X=['shape_20.tmp_0_slice_0']}, in_dtype = 2, op_device = , op_namescope = /, op_role = 0, op_role_var = [], out_dtype = 3, use_mkldnn = False, with_quant_attr = False)
    {Out=['tmp_181']} = less_than(inputs={X=['tmp_179'], Y=['tmp_180']}, axis = -1, force_cpu = False, op_device = , op_namescope = /, op_role = 0, op_role_var = [], with_quant_attr = False)
    {Out=['fill_constant_59.tmp_0']} = assign(inputs={X=['tmp_179']}, op_device = , op_namescope = /, op_role = 0, op_role_var = [], with_quant_attr = False)
    {Out=['undefined_var_30']} = assign(inputs={X=['_generated_var_30']}, op_device = , op_namescope = /, op_role = 0, op_role_var = [], with_quant_attr = False)
    {Out=['undefined_var_31']} = assign(inputs={X=['_generated_var_31']}, op_device = , op_namescope = /, op_role = 0, op_role_var = [], with_quant_attr = False)
    {Out=['undefined_var_32']} = assign(inputs={X=['range_11.tmp_0_slice_0']}, op_device = , op_namescope = /, op_role = 0, op_role_var = [], with_quant_attr = False)
    {Out=['fill_constant_61.tmp_0']} = assign(inputs={X=['_generated_var_32']}, op_device = , op_namescope = /, op_role = 0, op_role_var = [], with_quant_attr = False)
    {Out=['tmp_161']} = assign(inputs={X=['tmp_181']}, op_device = , op_namescope = /, op_role = 0, op_role_var = [], with_quant_attr = False)
}
```
The `select_input` for `_generated_var_28`, `_generated_var_24` will lead error like below:
```
Error occured at: /home/lvyongkang/benchmark/Paddle/paddle/fluid/ir_adaptor/translator/op_translator.cc:1871 :
Does not support rank inconsistency: dim1=1, dim2=0
```

#### Others
Pcard-67164

### CODE REVIEW

File: paddle/fluid/framework/new_executor/pir_adaptor/pir_adaptor_util.cc [hunk 1]
```diff
@@ -415,7 +415,25 @@ void HandleForSpecialOp(pir::Operation* op,
...
+      auto* t = var->GetMutable<phi::DenseTensor>();
+      if (op_name == paddle::dialect::DataOp::name()) {
+        auto shape = op->attribute<dialect::IntArrayAttribute>("shape");
+        auto dim = phi::make_ddim(shape.data().GetData());
+        auto dtype = op->attribute<dialect::DataTypeAttribute>("dtype");
+        auto place = op->attribute<dialect::PlaceAttribute>("place").data();
+        if (place.GetType() == phi::AllocationType::UNDEFINED) {
+          place = phi::CPUPlace();
+        }
+        if (phi::product(dim) >= 0) {
```
Comment (zhangbo9674) [inline paddle/fluid/framework/new_executor/pir_adaptor/pir_adaptor_util.cc:428]:
```suggestion
        if (!contain_unknown_dim(dim)) {
```

**Review by ceci3 — COMMENTED** [review]
look good for the test_dropout_op.py

File: paddle/fluid/ir_adaptor/translator/utils.cc [hunk 2]
```diff
@@ -105,5 +105,36 @@ std::vector<std::string> CheckUnregisteredOperation(
   return unregistered_ops;
 }
 
+phi::DataType PirTypeToPhiDType(pir::Type type) {
```
Comment (zyfncg) [inline paddle/fluid/ir_adaptor/translator/utils.cc:108]: 这个转换函数比较常用，是不可以放到一个公共目录下？

**Review by zyfncg — APPROVED** [review]

File: paddle/fluid/ir_adaptor/translator/utils.cc (hunk 2 above)
Comment (kangguangli) [reply paddle/fluid/ir_adaptor/translator/utils.cc:108]: 可以的，下个PR做一下，放到op_dialect里面吧

**Review by zhangbo9674 — APPROVED** [review]

**Pull request merged**